diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index e6852d9c2..fa3e13d4c 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -209,7 +209,7 @@ macro(resolve_dependency DEPENDENCY_NAME)
     set(ARG_IS_RUNTIME_DEPENDENCY TRUE)
   endif()
 
-  if(ARG_HAVE_ALT)
+  if(0) # ARG_HAVE_ALT)
     set(PACKAGE_NAME "${DEPENDENCY_NAME}Alt")
   else()
     set(PACKAGE_NAME ${DEPENDENCY_NAME})
