diff --git a/ThriftLibrary.cmake b/ThriftLibrary.cmake
index 0e0a2dbb56..62bf7bb6b1 100644
--- a/ThriftLibrary.cmake
+++ b/ThriftLibrary.cmake
@@ -202,6 +202,8 @@ macro(thrift_generate
     "THRIFT_INCLUDE_DIRECTORIES" # Multi-value args
     "${ARGN}")
 
+  find_program(THRIFT1 thrift1)
+
   set(thrift_include_directories)
   foreach(dir ${THRIFT_GENERATE_THRIFT_INCLUDE_DIRECTORIES})
     list(APPEND thrift_include_directories "-I" "${dir}")
